pkg/scheduler/internal/cache/snapshot_test.go

```diff
@@ -229,3 +230,107 @@ func TestCreateUsedPVCSet(t *testing.T) {
 		})
 	}
 }
+
+func TestNewSnapshot(t *testing.T) {
+	testCases := []struct {
+		name                         string
+		pods                         []*v1.Pod
+		nodes                        []*v1.Node
+		expectedNumNodes             int
+		expectedPodsWithAffinity     int
+		expectedPodsWithAntiAffinity int
+	}{
+		{
+			name:  "no pods no nodes",
+			pods:  nil,
+			nodes: nil,
+		},
+		{
+			name: "single pod single node",
+			pods: []*v1.Pod{
+				st.MakePod().Name("foo").Namespace("foo").Node("node-0").Obj(),
+			},
+			nodes: []*v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-0"},
+					Status:     v1.NodeStatus{},
+				},
+			},
+			expectedNumNodes: 1,
+		},
+		{
+			name: "multiple nodes, pod with affinity",
+			pods: []*v1.Pod{
+				st.MakePod().Name("foo").Namespace("ns").Node("node-0").Obj(),
+				st.MakePod().Name("bar").Namespace("ns").PodAffinity("baz", &metav1.LabelSelector{MatchLabels: map[string]string{"baz": "qux"}}, st.PodAffinityWithRequiredReq).Node("node-2").Obj(),
+			},
+			nodes: []*v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-0"},
+					Status:     v1.NodeStatus{},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-1"},
+					Status:     v1.NodeStatus{},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-2", Labels: map[string]string{"baz": "qux"}},
+					Status:     v1.NodeStatus{},
+				},
+			},
+			expectedNumNodes:         3,
+			expectedPodsWithAffinity: 1,
+		},
+		{
+			name: "multiple nodes, pod with affinity, pod with anti-affinity",
+			pods: []*v1.Pod{
+				st.MakePod().Name("foo").Namespace("ns").Volume(v1.Volume{}).Node("node-1").Obj(),
+				st.MakePod().Name("bar").Namespace("ns").PodAffinity("key", &metav1.LabelSelector{MatchLabels: map[string]string{"key": "value"}}, st.PodAffinityWithRequiredReq).Node("node-0").Obj(),
+				st.MakePod().Name("baz").Namespace("ns").PodAntiAffinity("another", &metav1.LabelSelector{MatchLabels: map[string]string{"another": "label"}}, st.PodAntiAffinityWithRequiredReq).Node("node-0").Obj(),
+			},
+			nodes: []*v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-0", Labels: map[string]string{"key": "value"}},
+					Status:     v1.NodeStatus{},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-1", Labels: map[string]string{"another": "label"}},
+					Status:     v1.NodeStatus{},
+				},
+			},
+			expectedNumNodes:             2,
+			expectedPodsWithAffinity:     1,
+			expectedPodsWithAntiAffinity: 1,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			snapshot := NewSnapshot(test.pods, test.nodes)
+			if len(test.nodes) == 0 {
+				info, err := snapshot.Get("dummy")
+				assert.Nil(t, info)
+				assert.Error(t, err)
+			} else {
+				info, err := snapshot.Get(test.nodes[0].Name)
+				assert.NotNil(t, info)
+				assert.NoError(t, err)
+			}
+
+			if len(test.pods) > 0 {
```

Comment:
If we add the `expectedUsedPVCSet` field, this is no longer needed. Let's verify the `expectedUsedPVCSet` directly.